src/main/java/com/artipie/api/artifactory/AddUpdatePermissionSlice.java

```diff
@@ -74,27 +67,21 @@ public AddUpdatePermissionSlice(final Settings settings) {
     @Override
     public Response response(final String line, final Iterable<Map.Entry<String, String>> headers,
         final Publisher<ByteBuffer> body) {
-        final Matcher matcher = AddUpdatePermissionSlice.PATH.matcher(
-            new RequestLineFrom(line).uri().toString()
-        );
-        final Response res;
-        if (matcher.matches()) {
-            res = new AsyncResponse(
+        final Optional<String> repoopt = new FromRqLine(line, FromRqLine.Group.REPO).get();
```

Comment:
@genryxy name `repoopt` looks odd, I'd suggest to name this variable simply `opt`